pandas/io/parquet.py

```diff
@@ -0,0 +1,160 @@
+""" parquet compat """
+
+from warnings import catch_warnings
+from pandas import DataFrame, RangeIndex, Int64Index, get_option
+from pandas.compat import range
+from pandas.io.common import get_filepath_or_buffer
+
+
+def get_engine(engine):
+    """ return our implementation """
+
+    if engine is None:
+        engine = get_option('io.parquet.engine')
+
+    if engine not in ['pyarrow', 'fastparquet']:
+        raise ValueError("engine must be one of 'pyarrow', 'fastparquet'")
+
+    if engine == 'pyarrow':
+        return PyArrowImpl()
+    elif engine == 'fastparquet':
+        return FastParquetImpl()
+
+
+class PyArrowImpl(object):
+
+    def __init__(self):
+        # since pandas is a dependency of pyarrow
+        # we need to import on first use
+
+        try:
+            import pyarrow
+            import pyarrow.parquet
+        except ImportError:
+            raise ImportError("pyarrow is required for parquet support\n\n"
+                              "you can install via conda\n"
+                              "conda install pyarrow -c conda-forge\n"
+                              "\nor via pip\n"
+                              "pip install pyarrow\n")
+        self.api = pyarrow
+
+    def write(self, df, path, compression=None, **kwargs):
```

Comment:
In most Parquet implementations, the default is to use `snappy` compression as the basic assumption is that Parquet file will be read quite often and mostly likely are not stored on the local machine thus `snappy` compression always saves more network transferral time as it cost in CPU time to decompress.